cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/BulkSparkConf.java

```diff
@@ -158,7 +157,7 @@ public class BulkSparkConf implements Serializable
     protected final String coordinatedWriteConfJson;
     private transient CoordinatedWriteConf coordinatedWriteConf; // it is transient; deserialized from coordinatedWriteConfJson in executors
 
-    public BulkSparkConf(SparkConf conf, Map<String, String> options)
+    public BulkSparkConf(SparkConf conf, Map<String, String> options, @Nullable Logger logger)
```

Comment:
NIT, I think preserving the existing constructor will avoid making a lot of changes in many calling sites.

```suggestion
    public BulkSparkConf(SparkConf conf, Map<String, String> options)
    {
        this(conf, options, null);
    }

    public BulkSparkConf(SparkConf conf, Map<String, String> options, @Nullable Logger logger)
```